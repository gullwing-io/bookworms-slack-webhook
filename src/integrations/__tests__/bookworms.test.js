import { jest } from "@jest/globals";
import {
  init,
  getBookmarks,
  listOfBookmarksMarkDown,
  findBookMarkFolder,
} from "../../integrations/bookworms.js";

// I rather not test against real data but mocks with native imports are causing me issues
await init(
  "https://raw.githubusercontent.com/thearegee/bookworms/main/demo/config/bookmarks.yaml"
);

describe("Bookworms", () => {
  describe("getBookmarks", () => {
    test("should return markdown for unfound folder", () => {
      expect(getBookmarks("junk")).toMatchSnapshot();
    });
    test("should return markdown for found folder", () => {
      expect(getBookmarks("folder 1")).toMatchSnapshot();
    });
  });
  describe("listOfBookmarksMarkDown", () => {
    test("should return all folders for commands", () => {
      expect(listOfBookmarksMarkDown()).toMatchSnapshot();
    });
  });
  describe("findBookMarkFolder", () => {
    test("should return found folder", () => {
      expect(
        findBookMarkFolder("folder 1", [
          {
            label: "folder 1",
            description: "This is to describe the sub folder structure",
            bookmarks: [
              {
                label: "sample url 1",
                description: "this is used to describe the bookmark",
                href: "https://www.mywebsite.com",
              },
            ],
          },
        ])
      ).toEqual({
        bookmarks: [
          {
            description: "this is used to describe the bookmark",
            href: "https://www.mywebsite.com",
            label: "sample url 1",
          },
        ],
        description: "This is to describe the sub folder structure",
        label: "folder 1",
      });
    });
    test("should return undefined if folder not found", () => {
      expect(
        findBookMarkFolder("junk", [
          {
            label: "folder 1",
            description: "This is to describe the sub folder structure",
            bookmarks: [
              {
                label: "sample url 1",
                description: "this is used to describe the bookmark",
                href: "https://www.mywebsite.com",
              },
            ],
          },
        ])
      ).toBeUndefined();
    });
  });
});
