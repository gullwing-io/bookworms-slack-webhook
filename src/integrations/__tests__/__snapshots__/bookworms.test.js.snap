// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookworms getBookmarks should return markdown for found folder 1`] = `
"


## folder 1

This is to describe the folder structure


### sub folder 1

This is to describe the sub folder structure

* [sample url 1](https://www.mywebsite.com) - this is used to describe the bookmark

_These bookmarks were last updated on 28/10/2022, 16:01:31 using [Bookworms](https://github.com/thearegee/bookworms)_"
`;

exports[`Bookworms getBookmarks should return markdown for unfound folder 1`] = `"Sorry junk could not be found, use \`/bookmarks all\` for a list of available bookmarks    "`;

exports[`Bookworms listOfBookmarksMarkDown should return all folders for commands 1`] = `
"Bookmarks are seperated into different domains, to get the specific bookmarks you can enter the following commands
    
\`/bookmarks folder 1\`
\`/bookmarks folder 2\`
    
    "
`;
